### DISCUSSION TITLE
Monorepo support?

### DISCUSSION THREAD

**Discussion by sairion** [biomejs/biome#779] [Ideas]
It seems Biome VS Code extension only uses `biome.json` at the workspace root, not 'the most adjacent' `biome.json` from the source which it lints, formats, etc. (I believe it is also Rome VS Code extension's behavior, after seeing https://github.com/rome/tools/issues/4206)

I'd like to know if support for monorepo (or, multiple `biome.json` in workspaces) is going to happen :)

**Comment by nhedger** [comment]
Hey @sairion,

Thanks for opening this issue.

I can't speak for everyone, but this is definitely something I'd like to see us support.

I'll move this issue to the https://github.com/biomejs/biome repository because that's where the logic should be implemented.

**Reply by castarco** [reply]
@nhedger I'm interested on this issue, as a Biome user, but also as a potential external contributor.

https://github.com/biomejs/biome/discussions/779#discussioncomment-9018799 .

**Comment by mkarbo** [comment]
We also use a monorepo, and we're blocked due to this issue as well.

We (would like to) have a biome.json at the repo root, which vscode & other extensions will load

**Comment by castarco** [comment]
Our particular case also favours the introduction of more monorepo-friendly features such as:
- Ensuring that the VSCode extension always applies the "closest" configuration file instead of having to declare overrides at the root level. There are some reasons for this:
  - security / auditability: including having struct rules enforcing "code owners" for certain parts of the repository
  - limiting the cascade effects of changing root files for monorepo tools (changing root files changes their hashes, and invalidates tons of "good" caches without any need)
- Applying the same principle as before for `biome` when executed from command line at the "root" level.

I still didn't read all the related issues, discussions, rfcs, etc. But I suspect that there is some work being performed on this front. What I don't know yet is if it will cover these specific issues. 

In case the ongoing work is covering a different set of issues, I'd like to discuss with the core developers if I can contribute code to solve the issues I presented.

**Comment by ematipico** [comment]
I am going to close this discussion, people can help and track progresses here: https://github.com/biomejs/biome/issues/2228
